cmd/signozschemamigrator/schema_migrator/traces_migrations.go

```diff
@@ -1,3 +1,360 @@
 package schemamigrator
 
 var TracesMigrations = []SchemaMigrationRecord{}
+
+// move them to TracesMigrations once it's ready to deploy
+var TracesMigrations_Staging = []SchemaMigrationRecord{
+	{
+		MigrationID: 1000,
+		UpItems: []Operation{
+			CreateTableOperation{
+				Database: "signoz_traces",
+				Table:    "signoz_index_v3",
+				Columns: []Column{
+					{Name: "ts_bucket_start", Type: ColumnTypeUInt64, Codec: "DoubleDelta, LZ4"},
+					{Name: "resource_fingerprint", Type: ColumnTypeString, Codec: "ZSTD(1)"},
+					{Name: "timestamp", Type: DateTime64ColumnType{Precision: 9}, Codec: "DoubleDelta, LZ4"},
+					{Name: "id", Type: FixedStringColumnType{Length: 27}, Codec: "ZSTD"},
+					{Name: "traceID", Type: FixedStringColumnType{Length: 32}, Codec: "ZSTD(1)"},
+					{Name: "spanID", Type: ColumnTypeString, Codec: "ZSTD(1)"},
+					{Name: "traceState", Type: ColumnTypeString, Codec: "ZSTD(1)"},
+					{Name: "parentSpanID", Type: ColumnTypeString, Codec: "ZSTD(1)"},
+					{Name: "flags", Type: ColumnTypeUInt32, Codec: "T64, ZSTD(1)"},
+					{Name: "name", Type: LowCardinalityColumnType{ColumnTypeString}, Codec: "ZSTD(1)"},
+					{Name: "kind", Type: ColumnTypeInt8, Codec: "T64, ZSTD(1)"},
+					{Name: "spanKind", Type: ColumnTypeString, Codec: "ZSTD(1)"},
+					{Name: "durationNano", Type: ColumnTypeUInt64, Codec: "T64, ZSTD(1)"},
+					{Name: "statusCode", Type: ColumnTypeInt16, Codec: "T64, ZSTD(1)"},
+					{Name: "statusMessage", Type: ColumnTypeString, Codec: "ZSTD(1)"},
+					{Name: "statusCodeString", Type: ColumnTypeString, Codec: "ZSTD(1)"},
+
+					{Name: "attributes_string", Type: MapColumnType{LowCardinalityColumnType{ColumnTypeString}, ColumnTypeString}, Codec: "ZSTD(1)"},
+					{Name: "attributes_number", Type: MapColumnType{LowCardinalityColumnType{ColumnTypeString}, ColumnTypeFloat64}, Codec: "ZSTD(1)"},
+					{Name: "attributes_bool", Type: MapColumnType{LowCardinalityColumnType{ColumnTypeString}, ColumnTypeBool}, Codec: "ZSTD(1)"},
+					{Name: "resources_string", Type: MapColumnType{LowCardinalityColumnType{ColumnTypeString}, ColumnTypeString}, Codec: "ZSTD(1)"},
+
+					{Name: "events", Type: ArrayColumnType{ColumnTypeString}, Codec: "ZSTD(2)"},
```

Comment:
How are we/users supposed to work with events?

Reply:
We just return the events in detail page, no search etc is allowed.

Reply:
Being able to easily query the event data is important: https://github.com/SigNoz/signoz/issues/2767. I believe we should think about it now since we are revamping the traces schema after a long time.

Reply:
Cool, but let's not block this PR as this anyway the changes won't move to production right away. We can finalize the changes before moving this production. I have attached the issues to trace improvements.